app/controllers/planning_applications_controller.rb

```diff
@@ -165,6 +168,19 @@ def determine
     redirect_to @planning_application
   end
 
+  def edit_constraints_form; end
+
+  def edit_constraints
+    constraint = params.reject! { |k, v| v.blank? || k.include?("-true") }.select { |k, _v| k.include?("constraint") }
+    constraint[params[:local].to_s] = true if params[:local]
+    if @planning_application.update(constraints: constraint.to_json)
```

Comment:
this will throw a rubocop error unless we use assign_attribute (it wants the update action to be called as update!)
An alternative if you cant use update! is to use assign_attributes(constraints) and then if @planning_application.save in the next line

Reply:
I also feel that this bit would be simplified by having a dedicated controller that accepts all these attributes as contraints params - in the building safety platform we did that here: 
https://github.com/unboxed/southwark-bsp/blob/main/app/controllers/surveys/building_external_wall_structures_controller.rb

thus allowing us to call all the params as "booleans" except one of the params "other" as a string, so instead of having all these transformations its true that a lot more text is written but then we can just call Constraints.update(contraints_params) 🤔  it feels like we are doing a lot of legwork instead of relying on the native rails methods

Reply:
![image](https://user-images.githubusercontent.com/9452321/112461401-cadbb680-8d57-11eb-85b2-373b58cb14ef.png)
params[:local] is already type string so we don't need the to_s